eventlet/patcher.py

```diff
@@ -424,17 +424,36 @@ def _fix_py2_rlock(rlock, tid):
         rlock._RLock__owner = tid
 
 
-def _fix_py3_rlock(old):
+def _fix_py3_rlock(old, tid):
     import gc
     import threading
+    from eventlet.green.thread import allocate_lock
     new = threading._PyRLock()
+    assert hasattr(new, "_block")
```

Comment:
We should not use `assert` outside of unit tests.

Indeed if someone his python interpreter with optimisation enabled (`python3.10 -o `), these statement would be removed.

Please consider raising an exception manually.

See the CPython doc:

https://github.com/python/cpython/blob/d9e1b5794a8fade21773d18f91a07f80b55c839c/Doc/using/cmdline.rst?plain=1#L315

Reply:
`patcher.py` is module loaded at runtime.

Reply:
> Indeed if someone use his python interpreter with optimisation enabled (python3.10 -o ), these statement would be removed.

I continue to be severely annoyed by that (C)Python design decision.